test/k8sT/DatapathConfiguration.go

```diff
@@ -304,6 +304,36 @@ var _ = Describe("K8sDatapathConfig", func() {
 	})
 
 	Context("DirectRouting", func() {
+		BeforeEach(func() {
+			switch {
+			case helpers.IsIntegration(helpers.CIIntegrationGKE):
+			default:
+				Skip("DirectRouting without AutoDirectNodeRoutes not supported")
+			}
+		})
+
+		It("Check connectivity with direct routing", func() {
+			deployCilium(map[string]string{
+				"global.tunnel":                 "disabled",
+				"global.k8s.requireIPv4PodCIDR": "true",
+				"global.endpointRoutes.enabled": "false",
+			})
+
+			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
+		})
+
+		It("Check connectivity with direct routing and endpointRoutes", func() {
+			deployCilium(map[string]string{
+				"global.tunnel":                 "disabled",
+				"global.k8s.requireIPv4PodCIDR": "true",
+				"global.endpointRoutes.enabled": "true",
+			})
+
+			Expect(testPodConnectivityAcrossNodes(kubectl)).Should(BeTrue(), "Connectivity test between nodes failed")
+		})
+	})
+
+	Context("AutoDirectNodeRoutes", func() {
 		BeforeEach(func() {
 			SkipIfIntegration(helpers.CIIntegrationFlannel)
 			SkipIfIntegration(helpers.CIIntegrationGKE)
```

Comment:
Does this still needs to be skipped?

Reply:
Yes, a single L2 network is not given. In fact, auto-direct-node routes is not required at all on GKE as the default route will point to a router that can route all PodIPs as announced in Node.Spec.PodCIDR[s] so we can simply disable tunneling to achieve native routing across everything.